endpoints/openrtb2/video_auction.go

```diff
@@ -147,13 +147,26 @@ func (deps *endpointDeps) VideoAuctionEndpoint(w http.ResponseWriter, r *http.Re
 	//create full open rtb req from full video request
 	mergeData(videoBidReq, bidReq)
 
+	initialPodNumber := len(videoBidReq.PodConfig.Pods)
+	if len(podErrors) > 0 {
+		//remove incorrect pods
+		videoBidReq = cleanupVideoBidRequest(videoBidReq, podErrors)
+	}
+
 	//create impressions array
-	imps, errs := deps.createImpressions(videoBidReq)
-	if errs != nil {
-		errL = append(errL, errs...)
+	imps, podErrors := deps.createImpressions(videoBidReq, podErrors)
+
+	if len(podErrors) == initialPodNumber {
```

Comment:
Wouldn't it make more sense to check `len(imps) == 0` here? The `initialPodNumber` could be higher here as compared to that of the request based on which pods were removed above in line 153, no?